bundles/org.openhab.binding.modbus.e3dc/src/main/java/org/openhab/binding/modbus/e3dc/internal/handler/E3DCThingHandler.java

```diff
@@ -0,0 +1,453 @@
+        string2WattChannel = channelUID(thing, STRINGS_GROUP, STRING2_DC_OUTPUT_CHANNEL);
+        string3AmpereChannel = channelUID(thing, STRINGS_GROUP, STRING3_DC_CURRENT_CHANNEL);
+        string3VoltChannel = channelUID(thing, STRINGS_GROUP, STRING3_DC_VOLTAGE_CHANNEL);
+        string3WattChannel = channelUID(thing, STRINGS_GROUP, STRING3_DC_OUTPUT_CHANNEL);
+    }
+
+    public @Nullable ModbusCommunicationInterface getComms() {
+        return comms;
+    }
+
+    public int getSlaveId() {
+        return slaveId;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // no control of E3DC device possible yet
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+        scheduler.execute(() -> {
+            E3DCConfiguration localConfig = getConfigAs(E3DCConfiguration.class);
+            config = localConfig;
+            ModbusCommunicationInterface localComms = connectEndpoint();
+            if (localComms != null) {
+                // register low speed info poller
+                ModbusReadRequestBlueprint infoRequest = new ModbusReadRequestBlueprint(slaveId,
+                        ModbusReadFunctionCode.READ_MULTIPLE_REGISTERS, INFO_REG_START, INFO_REG_SIZE, 3);
+                infoPoller = localComms.registerRegularPoll(infoRequest, INFO_POLL_REFRESH_TIME_MS, 0,
+                        this::handleInfoResult, this::handleInfoFailure);
+
+                ModbusReadRequestBlueprint dataRequest = new ModbusReadRequestBlueprint(slaveId,
+                        ModbusReadFunctionCode.READ_MULTIPLE_REGISTERS, POWER_REG_START,
+                        REGISTER_LENGTH - INFO_REG_SIZE, 3);
+                if (config != null) {
+                    dataPoller = localComms.registerRegularPoll(dataRequest, localConfig.refresh, 0,
+                            this::handleDataResult, this::handleDataFailure);
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            "E3DC Configuration missing");
+                }
+            } // else state handling performed in connectEndPoint function
+        });
+    }
+
+    /**
+     * Get a reference to the modbus endpoint
+     */
+    private @Nullable ModbusCommunicationInterface connectEndpoint() {
+        if (comms != null) {
+            return comms;
+        }
+
+        ModbusEndpointThingHandler slaveEndpointThingHandler = getEndpointThingHandler();
+        if (slaveEndpointThingHandler == null) {
+            @SuppressWarnings("null")
+            String label = Optional.ofNullable(getBridge()).map(b -> b.getLabel()).orElse("<null>");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE,
+                    String.format("Bridge '%s' is offline", label));
+            logger.debug("No bridge handler available -- aborting init for {}", label);
```

Comment:
Do you need the additional log message, as the framework already logs the message in `updateStatus()`? Same for below.

Reply:
I took over the code from @ssalonen and I didn't took a closer look on the debug messages so far. Now I removed additional debug message which are updated in the thing status